src/remote_cluster.rs

```diff
@@ -0,0 +1,237 @@
+            active_bucket: Mutex::new(resources.active_bucket),
+            active_scope: Mutex::new(resources.active_scope),
+            active_collection: Mutex::new(resources.active_collection),
+            tls_config,
+            timeouts: Mutex::new(timeouts),
+            capella_org,
+            kv_batch_size,
+            cluster_type,
+        }
+    }
+
+    pub fn cluster(&self) -> Arc<Client> {
+        let mut c = self.cluster.lock().unwrap();
+        if c.is_none() {
+            *c = Some(Arc::new(Client::new(
+                self.hostnames.clone(),
+                self.username.clone(),
+                self.password.clone(),
+                self.tls_config.clone(),
+            )));
+        }
+        c.as_ref().unwrap().clone()
+    }
+
+    pub fn active_bucket(&self) -> Option<String> {
+        self.active_bucket.lock().unwrap().as_ref().cloned()
+    }
+
+    pub fn set_active_bucket(&self, name: String) {
+        let mut active = self.active_bucket.lock().unwrap();
+        *active = Some(name);
+    }
+
+    pub fn active_scope(&self) -> Option<String> {
+        self.active_scope.lock().unwrap().as_ref().cloned()
+    }
+
+    pub fn set_active_scope(&self, name: String) {
+        let mut active = self.active_scope.lock().unwrap();
+        *active = Some(name);
+    }
+
+    pub fn active_collection(&self) -> Option<String> {
+        self.active_collection.lock().unwrap().as_ref().cloned()
+    }
+
+    pub fn set_active_collection(&self, name: String) {
+        let mut active = self.active_collection.lock().unwrap();
+        *active = Some(name);
+    }
+
+    pub fn deactivate(&self) {
+        let mut c = self.cluster.lock().unwrap();
+        if c.is_some() {
+            *c = None;
+        }
+    }
+
+    pub fn hostnames(&self) -> &Vec<String> {
+        &self.hostnames
+    }
+
+    pub fn username(&self) -> &str {
+        self.username.as_str()
+    }
+
+    pub fn password(&self) -> &str {
+        self.password.as_str()
+    }
+
+    pub fn tls_config(&self) -> &ClusterTlsConfig {
+        &self.tls_config
+    }
+
+    pub fn timeouts(&self) -> ClusterTimeouts {
+        let active = self.timeouts.lock().unwrap();
+        active.clone()
+    }
+
+    pub fn set_timeouts(&self, timeouts: ClusterTimeouts) {
+        let mut active = self.timeouts.lock().unwrap();
+        *active = timeouts
+    }
+
+    pub fn capella_org(&self) -> Option<String> {
+        self.capella_org.clone()
+    }
+
+    pub fn kv_batch_size(&self) -> u32 {
+        self.kv_batch_size
+    }
+
+    #[allow(dead_code)]
+    pub fn cluster_type(&self) -> RemoteClusterType {
+        self.cluster_type
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct ClusterTimeouts {
+    data_timeout: Duration,
+    query_timeout: Duration,
+    analytics_timeout: Duration,
+    search_timeout: Duration,
+    management_timeout: Duration,
+}
+
+impl Default for ClusterTimeouts {
+    fn default() -> Self {
```

Comment:
This makes me wonder, do we have configuration profiles here?

Reply:
No we don't. I think we easily could do in the future though. We could even automatically switch the default profile depending on cluster type. https://github.com/couchbaselabs/couchbase-shell/issues/228 filed.